core/message_queue.go

```diff
@@ -174,6 +183,30 @@ func (mq *MessageQueue) Size() int64 {
 	return l
 }
 
+// Oldest returns the oldest message stamp in the MessageQueue.
+// Oldest returns 0 if the queue is emptystarts
+func (mq *MessageQueue) Oldest() uint64 {
```

Comment:
Don't export this method without a reason to. Testing is an ok reason, but document this with "Exported for testing".

Note that this is a very neat case where what we really want to test is that the metric has the right value. If the metric registry was injected, we could do this directly. Let this be a prompt to follow through with that work!